### DISCUSSION TITLE
QTable: Emit an event on col sort

### DISCUSSION THREAD

**Discussion by nlassaux** [quasarframework/quasar#17550] [General - Components / Directives / etc] [answered]
Hi there! 

I'm trying to detect when a column has been sorted by a user. I was expecting an event for it, but I can't seem to find any.
Is there any workaround you'd see to accomplish it?

Thank you!

**Comment by yusufkandemir** [comment, marked as answer]
You can use `@update:pagination`, but it won't only be triggered on sorting changes. You can use a separate variable to check if it was changed, if you need such logic.
```html
<q-table
  @update:pagination="onPagination"
...

<script setup>
function onPagination({ sortBy, descending }) {
  console.log({ sortBy, descending })
}
</script>
```
Alternatively, you can use `v-model:pagination` to keep it in sync, then use `watch` to check if it has been changed:
```html
<q-table
  v-model:pagination="pagination"
...

<script setup>
const pagination = ref({})

watch(
  [() => pagination.value.sortBy, () => pagination.value.descending],
  ([sortBy, descending], [oldSortBy, oldDescending]) => {
    console.log({ sortBy, descending })
  }
)
</script>
```
